_enumerable.rb

```diff
@@ -0,0 +1,84 @@
+module Enumerable
+  def my_each
+    each do |ele|
+      yield ele
+    end
+    self
+  end
+
+  def my_each_with_index
+    length.times do |indx|
+      yield self[indx], indx
+    end
+    self
+  end
+
+  def my_select
+    selected_elements = []
+    my_each do |k|
+      selected_elements.push(k) if yield k
+    end
+    selected_elements
+  end
+
+  def my_all?
+    my_each do |ele|
+      # returns false if an element that doesn't satsify the condition is found
+      return false unless yield ele
+    end
+    true
+  end
+
+  def my_any?
+    my_each do |ele|
+      return true if yield ele
+    end
+    false
+  end
+
+  def my_none?
+    my_each do |ele|
+      return false if yield ele
+    end
+    true
+  end
```

Comment:
Is expected that: 

#my_none? when no block or argument is given returns true only if none of the collection members is true

#my_none? when no block or argument is given returns false only if one of the collection members is true

#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class

#my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric

#my_none? when a class is passed as an argument returns false if any of the collection is a member of such class

#my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex

#my_none? when a Regex is passed as an argument returns false if any of the collection matches the Regex

#my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern

#my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern